docs/maven-example.md

```diff
@@ -0,0 +1,74 @@
+# Fabric8 Maven Plugin + Kompose:
+Lets deploy a Springboot Java application with Docker Compose file using Fabric8 Maven Plugin to Kubernetes or OpenShift.
+##### Requirements
+* Linux or MacOS or Windows
+* JDK 1.7+ - [JDK Quick Installation Guide](http://openjdk.java.net/install/)
+* Maven 3.x+ - [Maven Installation Guide](https://maven.apache.org/install.html) 
+* Kompose
+
+__1. Clone the example project from GitHub__
+```bash
+$ git clone https://github.com/piyush1594/kompose-maven-example.git
+```
+Navigate (cd) to `kompose-maven-example` directory. Use command
+```bash
+$ cd kompose-maven-example
+```
+__2. Install Kompose through Maven__ 
+```
+$ mvn fabric8:install
+```
+This command installs the `kompose` and 'fabric8' on the host.
+
+__3. Add Fabric8 Maven Plugin to project__     
+```
+$ mvn io.fabric8:fabric8-maven-plugin:3.5.28:setup
+```
+Adds the Fabric8 Maven Plugin configuration to `pom.xml` of project. `pom.xml` is manifest or deployment descriptor file of maven project.
+
+__4. Configure Fabric8 Maven Plugin to use Docker Compose file__
+```
+<plugin>
+  <groupId>io.fabric8</groupId>
+  <artifactId>fabric8-maven-plugin</artifactId>
+
+  <!-- ... -->
+
+  <configuration>
+    <composeFile>path for docker compose file</composeFile>
+  </configuration>
+
+  <executions>
+    <execution>
+      <goals>
+        <goal>resource</goal>
+        <goal>build</goal>
+      </goals>
+    </execution>
+  </executions>
+</plugin>
+```
+Add the `<configuration>` and `<executions>` sections to `pom.xml` as shown in above `pom.xml` snippet. Update the `<composeFile>` to provide the relative path of Docker Compose file from `pom.xml`
+
+__5. Deploy application on Kubernetes or OpenShift__
+ Make sure that Kubernetes/OpenShift cluster or Minikube/minishift is running. In case, if anything of this is not running, you can run Minikube to test this application by using following command. 
+```bash
+$ minikube start
+```
+ Below command deploys this application on Kubernetes or OpenShift.
+```bash
+$ mvn fabric8:deploy  
+```
+
+Now that your service has been deployed, let's access it by querying `pod`, `service` from Kubernetes or OpenShift.
+```bash
+$ kubectl get pods
+NAME                                    READY     STATUS      RESTARTS   AGE
+springboot-docker-compose-1-vk80j       1/1       Running     0          5m
+springboot-docker-compose-s2i-1-build   0/1       Completed   0          7m
+```
+```bash
+$ kubectl get svc
+NAME                        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+springboot-docker-compose   172.30.205.137   <none>        8080/TCP   6m
+```
```

Comment:
you need to add a tutorial on how to actually access these... bringing up the web interface aka `minikube service foobar` or `minishift openshift service foobar` if a route has been made.